src/controllers/courseController.js

```diff
@@ -0,0 +1,102 @@
+require('dotenv').config();
+const admin = require('../config/firebase');
+const db = admin.firestore();
+const axios = require('axios');
+const CourseModel = require('../models/CourseModel'); 
+
+// Create a new course
+exports.createCourse = async (req, res) => {
+    const courseData = req.body;
+   
+    try {
+        const newCourse = new CourseModel(courseData);
+        const courseRef = db.collection('CoursesDetails').doc(); // Generate a new document ID
+        await courseRef.set(newCourse.toFirestore());
+
+        res.status(201).send({ message: 'Course created successfully', courseId: courseRef.id });
+    } catch (error) {
+        console.error('Error creating course:', error);
+        res.status(400).send({ error: error.message });
+    }
+};
+
+// Fetch a specific course by ID
+exports.getCourse = async (req, res) => {
+    const { uid } = req.params;
+
+    try {
+        const courseDoc = await db.collection('CoursesDetails').doc(uid).get();
+
+        if (!courseDoc.exists) {
+            return res.status(404).send({ message: 'Course not found' });
+        }
+
+        const courseData = CourseModel.fromFirestore(courseDoc.data());
+        res.status(200).send(courseData);
+    } catch (error) {
+        console.error('Error fetching course:', error);
+        res.status(400).send({ error: error.message });
+    }
+};
+
+// Fetch all courses
+exports.getAllCourses = async (req, res) => {
+    try {
+        const coursesSnapshot = await db.collection('CoursesDetails').get();
+
+        if (coursesSnapshot.empty) {
+            return res.status(404).send({ message: 'No courses found' });
```

Comment:
We don't need this check. A 404 would be if I am asking for a specific course that isn't real. If I want all and there just happen to be none, that's not an error there just aren't any courses. For the frontend, we would expect to get an empty array. The request worked, there's just no courses we need to show on the dashboard.